add_script, get_script, get_scripts

superpowers04 opened the pull request.

superpowers04 committed:
add_script, get_script, get_scripts

superpowers04 committed:
Fix addscript and getscript using broken pathing

superpowers04 committed:
Actually, that broke addscript

TheBunnyMan123 commented:
I honestly want this just for get_script. Not having to do cheeky avatar NBT stuff would be great

superpowers04 committed:
Fix addscript

i am dumb and apparently copied the wrong thing into the thing, you know how it is

superpowers04 committed:
Fix addscript for the last time(for real)

dumb mf forgor that figura uses 2 paths for scripts

UnlikePaladin commented:
I'm still not sure how I feel about add script but get script and get scripts I'm fine with

UnlikePaladin commented:
hm after some further discussion, it should be fine I guess, don't go nuts with it though, was it tested with require() though? Just check the dev channel

UnlikePaladin commented on common/src/main/java/org/figuramc/figura/lua/FiguraLuaRuntime.java:
```diff
@@ -149,6 +150,11 @@ private void loadExtraLibraries() {
         this.setGlobal("loadstring", loadstring);
 
+        // addScript & getScript
+        this.setGlobal("add_script", addScript);
```
camelcase pls

UnlikePaladin commented:
I'll merge it, just change that to camelCase pls lol

auriafoxgirl commented:
Meow

superpowers04 committed:
Update FiguraLuaRuntime.java

UnlikePaladin reviewed (commented):
lgtm
